srml/council/src/lib.rs

```diff
@@ -14,7 +14,159 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Council system: Handles the voting in and maintenance of council members.
+//! # Council Module
+//!
+//! The Council module provides tools to manage the council and proposals.
+//!
+//! - **Seats**
+//! 	- [`seats::Trait`](./seats/trait.Trait.html)
+//! 	- [`Call`](./seats/enum.Call.html)
+//! 	- [`Module`](./seats/struct.Module.html)
+//! - **Motions**
+//! 	- [`motions::Trait`](./motions/trait.Trait.html)
+//! 	- [`Call`](./motions/enum.Call.html)
+//! 	- [`Module`](./motions/struct.Module.html)
+//! - **Voting**
+//! 	- [`voting::Trait`](./voting/trait.Trait.html)
+//! 	- [`Call`](./voting/enum.Call.html)
+//! 	- [`Module`](./voting/struct.Module.html)
+//!
+//! ## Overview
+//!
+//! The Council module provides functionality to handle:
+//!
+//! - The voting in and maintenance of council members.
+//! - Proposing, vetoing, and passing motions.
+//!
+//! The Council's role is to represent passive stakeholders. The Council is an on-chain entity comprised of
+//! a set of account IDs. Its primary tasks are to propose sensible referenda and thwart any uncontroversially
+//! dangerous or malicious referenda.
+//!
+//! ### Terminology
+//!
+//! #### Council Proposals
+//!
+//! - **Council motion:** A mechanism used to enact a proposal.
+//! - **Council origin:** The council (not root) that contains the council motion mechanism.
```

Comment:
What else do we know about this --apparently-- origin type? I don't see it being used anywhere else in the doc. If I am missing it, would be nice to link here to that reference point.

Reply:
For some reason I thought there was a special impl of origin, but only for checking if an origin contains `n` council members. All the external calls don't check origin, so I think it's best to remove.

Reply:
`CouncilTrait` actually does accept a `type Origin: From<Origin>;` so maybe worth looking into it first. I don't know how this related to the `System::Trait`'s `Orgin` though.

Outside the scope of this PR but I think the general concept of `origin` is worth a separate doc so that ppl can link to it when needed (cc/ping @shawntabrizi).